subsys/secure_storage/src/its/transform/aead.c

```diff
@@ -0,0 +1,122 @@
+/* Copyright (c) 2024
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <../library/psa_crypto_driver_wrappers.h>
+#include <zephyr/secure_storage/its/transform.h>
+#include <zephyr/secure_storage/its/transform/aead_get.h>
+
+static psa_status_t psa_aead_crypt(psa_key_usage_t operation, secure_storage_its_uid_t uid,
+				   const uint8_t nonce
+				   [static CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_NONCE_SIZE],
+				   size_t add_data_len, const void *add_data, size_t input_len,
+				   const void *input, size_t output_size, void *output,
+				   size_t *output_len)
+{
+	psa_status_t ret;
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+	uint8_t key[CONFIG_SECURE_STORAGE_ITS_TRANSFORM_AEAD_KEY_SIZE];
+	psa_key_type_t key_type;
+	psa_algorithm_t alg;
+	psa_status_t (*aead_crypt)(const psa_key_attributes_t *attributes, const uint8_t *key,
+				   size_t key_size, psa_algorithm_t alg, const uint8_t *nonce,
+				   size_t nonce_length, const uint8_t *add_data,
+				   size_t add_data_len, const uint8_t *input, size_t input_len,
+				   uint8_t *output, size_t output_size, size_t *output_len);
+
+	secure_storage_its_transform_aead_get_scheme(&key_type, &alg);
+
+	psa_set_key_usage_flags(&key_attributes, operation);
+	psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_type(&key_attributes, key_type);
+	psa_set_key_algorithm(&key_attributes, alg);
+	psa_set_key_bits(&key_attributes, sizeof(key) * 8);
+
+	aead_crypt = (operation == PSA_KEY_USAGE_ENCRYPT) ?
```

Comment:
Why are you calling directly into drivers' wrappers? IMO you should call `psa_aead_encrypt/psa_aead_decrypt` instead.